File: week9/lab_java/bank/Manager.java
package bank;

public class Manager{
    private String name;
    private Bank bank;
    
    public Manager(Bank bank)    {
       this.name = "John Smith";
       this.bank = bank;
    }

    private void view(){
        this.bank.view();
    }
    
    private void show(){
        this.bank.show();
    }
    
    private void remove(){
        this.bank.remove();
    }
    
    private void add(){
        this.bank.add();
    }
    
    private char readChoice(){
        System.out.print("Manager menu (a/r/s/v/x): ");
        return Config.IN.next().charAt(0);
    }

    public void use() {
        System.out.println(name+" admin menu: "+Config.DTF.format(Config.NOW));
        char c;
        while((c = readChoice()) != 'x'){
            switch(c){
                case 'a': this.add(); break;
                case 'r': this.remove(); break;
                case 's': this.show(); break;
                case 'v': this.view(); break;
                default: help(); break;
            }
        }
        System.out.println("Back to Bank menu");
    }
    
    private void help() {
        System.out.println("Menu options");
        System.out.println("a = add a new customer");
        System.out.println("r = remove a customer");
        System.out.println("s = show the selected customer statement");
        System.out.println("v = view all customers");
        System.out.println("x = exit");
    }
}

File: week9/lab_py/bank/manager.py
from config import DTF,NOW

class Manager:
    def __init__(self,bank):
        self.name = "John Smith"
        self.bank = bank

    def view(self):
        self.bank.view()

    def show(self):
        self.bank.show()

    def remove(self):
        self.bank.remove()

    def add(self):
        self.bank.add()

    def read_choice(self):
        print("Manager menu (a/r/v/s/x): ", end="")
        return input().strip().lower()
    
    def use(self):
        print(f'{self.name} admin menu: {NOW.strftime(DTF)}')
        c = self.read_choice()
        while c!='x':
            match c:
                case 'a':
                    self.add()
                case 'r':
                    self.remove()
                case 's':
                    self.show()
                case 'v':
                    self.view()
                case _:
                    self.help()
            c = self.read_choice()
        print("Back to Bank menu")

    def help(self):
        print("Menu options")
        print("a = add a customer")
        print("r = remove a customer")
        print("v = view all customers")
        print("s = show the selected customer")
        print("x = exit")